src/services/data_extractor_service.py

```
@@ -1,75 +1,84 @@
-            
+
             # 列名の確認
-            expected_columns = ['time', 'value']
-            if list(df.columns[:2]) != expected_columns:
-                raise ValueError(f"1行目は 'time', 'value' という列名である必要があります。現在: {list(df.columns[:2])}")
-            
+            required = {"time", "value"}
+            if not required.issubset(set(df.columns)):
+                raise ValueError(
+                    f"列名に 'time', 'value' が必要です。現在: {list(df.columns)}"
+                )
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)